algo/src/main/java/com/hartwig/actin/algo/evaluation/general/HasMaximumBMI.java

```diff
@@ -0,0 +1,69 @@
+package com.hartwig.actin.algo.evaluation.general;
+
+import com.hartwig.actin.PatientRecord;
+import com.hartwig.actin.algo.datamodel.Evaluation;
+import com.hartwig.actin.algo.datamodel.EvaluationResult;
+import com.hartwig.actin.algo.datamodel.ImmutableEvaluation;
+import com.hartwig.actin.algo.evaluation.EvaluationFactory;
+import com.hartwig.actin.algo.evaluation.EvaluationFunction;
+import com.hartwig.actin.clinical.datamodel.BodyWeight;
+import com.hartwig.actin.clinical.sort.BodyWeightDescendingDateComparator;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Optional;
+
+public class HasMaximumBMI implements EvaluationFunction {
+
+    static final String EXPECTED_UNIT = "kilogram";
+    static final double MIN_EXPECTED_HEIGHT_METRES = 1.5;
+    static final double MAX_EXPECTED_HEIGHT_METRES = 2.0;
+
+    private final int maximumBMI;
+
+    HasMaximumBMI(final int maximumBMI) {
+        this.maximumBMI = maximumBMI;
+    }
+
+    private double calculateWeightForBmiAndHeight(double bmi, double height) {
```

Comment:
I generally put private functions below the public evaluate function, feels more logical in terms of ordering. Also, is there a reason why this function is non-static?